vmware-tanzu/tanzu-framework
# Update plugin templates to support Go 1.17

### What this PR does / why we need it

Updates go.mod template to support Go 1.17 and latest tanzu-framework.
Updates CI files to use Go 1.17 for builds.

### Which issue(s) this PR fixes


Fixes #857 

### Describe testing done for PR

Verify new plugin can be created and successfully built.

1. Create new plugin:

```
$ tanzu builder init test-plugin --repo-type=github
$ tanzu builder cli add-plugin test-plugin
$ make init 
$ git add -A && git commit -m "Initial commit"
```
2. Build plugin:

```
$ make build-install-local
tanzu builder cli compile --version v0.0.1 --ldflags "-X 'github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli.BuildDate=2022-01-25' -X 'github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli.BuildSHA=de74ab9' -X 'github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli.BuildVersion=v0.0.1'" --path ./cmd/plugin --target local --artifacts artifacts/darwin/amd64/cli
ℹ  building local repository at ./artifacts/darwin/amd64/cli
ℹ  🐯 - building plugin at path "cmd/plugin/test-plugin"
ℹ  🐯 - $ /usr/local/bin/go build -o /Users/ragasthya/go/src/github.com/vmware-tanzu/tanzu-framework/test-plugin/artifacts/darwin/amd64/cli/test-plugin/v0.0.1/tanzu-test-plugin-darwin_amd64 -ldflags -X 'github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli.BuildDate=2022-01-25' -X 'github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli.BuildSHA=de74ab9' -X 'github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli.BuildVersion=v0.0.1' -tags  ./cmd/plugin/test-plugin
ℹ  🐯 - $ /usr/local/bin/go build -o /Users/ragasthya/go/src/github.com/vmware-tanzu/tanzu-framework/test-plugin/artifacts/darwin/amd64/cli/test-plugin/v0.0.1/test/tanzu-test-plugin-test-darwin_amd64 -ldflags -X 'github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli.BuildDate=2022-01-25' -X 'github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli.BuildSHA=de74ab9' -X 'github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli.BuildVersion=v0.0.1' -tags  ./cmd/plugin/test-plugin/test
========
✔  successfully built local repository
tanzu builder publish --type local --plugins ""test-plugin"" --version v0.0.1 --os-arch "darwin-amd64" --local-output-discovery-dir "./artifacts/published/darwin-amd64/discovery/standalone" --local-output-distribution-dir "./artifacts/published/darwin-amd64/distribution" --input-artifact-dir ./artifacts
tanzu plugin install all --local ./artifacts/published/darwin-amd64
Installing plugin 'test-plugin:v0.0.1'
✔  successfully installed 'all' plugin
```

### Release note

```release-note
New plugins created using `tanzu builder` command now supports Go 1.17 and includes the latest `tanzu-framework` release as dependency.
```

### PR Checklist



      


### Additional information

#### Special notes for your reviewer
